tests/Unit/Evolution/Systems/ScalarWave/Test_Characteristics.cpp

```diff
@@ -0,0 +1,388 @@
+  // Evaluate analytic solution
+  const auto vars = solution.variables(
+      x, t,
+      tmpl::list<ScalarWave::Pi, ScalarWave::Phi<Dim>, ScalarWave::Psi>{});
+  // Get ingredients
+  const size_t n_pts = x.begin()->size();
+  const auto gamma_2 = make_with_value<Scalar<DataVector>>(x, 0.1);
+  const auto& psi = get<ScalarWave::Psi>(vars);
+  const auto& phi = get<ScalarWave::Phi<Dim>>(vars);
+  const auto& pi = get<ScalarWave::Pi>(vars);
+  // Outward 3-normal to the surface on which characteristic fields are needed
+  auto unit_normal_one_form =
+      make_with_value<tnsr::i<DataVector, Dim, Frame::Inertial>>(x, 1.);
+  const auto norm_of_one_form = magnitude(unit_normal_one_form);
+  for (size_t i = 0; i < Dim; ++i) {
+    unit_normal_one_form.get(i) /= get(norm_of_one_form);
+  }
+
+  // Compute characteristic fields locally
+  Scalar<DataVector> phi_dot_normal{DataVector(n_pts, 0.)};
+  for (size_t i = 0; i < Dim; ++i) {
+    get(phi_dot_normal) += unit_normal_one_form.get(i) * phi.get(i);
+  }
+
+  tnsr::i<DataVector, Dim, Frame::Inertial> phi_dot_projection_tensor{
+      DataVector(n_pts, 0.)};
+  for (size_t i = 0; i < Dim; ++i) {
+    phi_dot_projection_tensor.get(i) =
+        phi.get(i) - unit_normal_one_form.get(i) * get(phi_dot_normal);
+  }
+
+  const auto& upsi = psi;
+  const auto& uzero = phi_dot_projection_tensor;
+  const Scalar<DataVector> uplus{get(pi) + get(phi_dot_normal) -
+                                 get(gamma_2) * get(psi)};
+  const Scalar<DataVector> uminus{get(pi) - get(phi_dot_normal) -
+                                  get(gamma_2) * get(psi)};
+
+  // Check that locally computed fields match returned ones
+  const auto uvars = ScalarWave::CharacteristicFieldsCompute<Dim>::function(
+      gamma_2, psi, pi, phi, unit_normal_one_form);
+
+  const auto& upsi_from_func = get<ScalarWave::Tags::UPsi>(uvars);
+  const auto& uzero_from_func = get<ScalarWave::Tags::UZero<Dim>>(uvars);
+  const auto& uplus_from_func = get<ScalarWave::Tags::UPlus>(uvars);
+  const auto& uminus_from_func = get<ScalarWave::Tags::UMinus>(uvars);
+
+  CHECK_ITERABLE_APPROX(upsi, upsi_from_func);
+  CHECK_ITERABLE_APPROX(uzero, uzero_from_func);
+  CHECK_ITERABLE_APPROX(uplus, uplus_from_func);
+  CHECK_ITERABLE_APPROX(uminus, uminus_from_func);
+}
+}  // namespace
+
+namespace {
+template <typename Tag, size_t Dim>
+typename Tag::type evol_field_with_tag(
+    const Scalar<DataVector>& gamma_2, const Scalar<DataVector>& u_psi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& u_zero,
+    const Scalar<DataVector>& u_plus, const Scalar<DataVector>& u_minus,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& unit_normal_one_form) {
+  return get<Tag>(
+      ScalarWave::EvolvedFieldsFromCharacteristicFieldsCompute<Dim>::function(
+          gamma_2, u_psi, u_zero, u_plus, u_minus, unit_normal_one_form));
+}
+
+template <size_t Dim>
+void test_evolved_from_characteristic_fields() noexcept {
+  const DataVector used_for_size(20);
```

Comment:
Why not the usual value, 5?

Reply:
I didn't know we had a usual value, changed to 5 now :)